src/pipeline_manager.py

```diff
@@ -37,6 +35,16 @@ def train(pipeline_name, dev_mode):
     annotations = pd.read_csv(PARAMS.annotations_filepath)
     annotations_human_labels = pd.read_csv(PARAMS.annotations_human_labels_filepath)
 
+    if DESIRED_CLASS_SUBSET:
+        LOGGER.info("Training on a reduced class subset: {}".format(DESIRED_CLASS_SUBSET))
+        annotations = reduce_number_of_classes(annotations,
+                                               DESIRED_CLASS_SUBSET,
+                                               PARAMS.class_mappings_filepath)
+
+        annotations_human_labels = reduce_number_of_classes(annotations_human_labels,
+                                                            DESIRED_CLASS_SUBSET,
+                                                            PARAMS.class_mappings_filepath)
+
     if PARAMS.default_valid_ids:
```

Comment:
@i008 what happens when the validation ids sample does not contain any of the subset classes. 
Should we simple not use dev_mode with class subset? If so let's fail with bam.

Reply:
You are right, did not realize that. Anyhow i dont think either-or is a good option. I think we should have full functionality both ways. So if SUBSET we should sample the vali_ids after filtering them